algviz/parser/structures.py

```diff
@@ -228,7 +251,11 @@ def __eq__(self, other):
                 self.data == other.data and
```

Comment:
This comment is actually for the line above, but because it isn't a new line or whatever it won't let me comment on it. That line being "return (super().__eq__(other) and". The parent class's definition of __eq__ simply looks to see if the uids are the same, but it's very well possible you have two trees that are different instances perhaps (and different UIDs as such), but are the same tree. 
I could be mistaken here though, as this might be intended.